watertap/tools/oli_api/flash.py

```diff
@@ -75,420 +73,1163 @@
+                )
 
-    def _build_input_list(self, state_vars):
-        """
-        Build input list for wateranalysis flash method.
+        input_list.extend(
+            [
+                {
+                    "group": "Calculation Options",
+                    "name": "AllowSolidsToForm",
+                    "value": bool(allow_solids),
+                },
+                {
+                    "group": "Calculation Options",
+                    "name": "CalcAlkalnity",
+                    "value": bool(calc_alkalinity),
+                },
+            ]
+        )
+        conc_unit = self.input_unit_set["molecularConcentration"]["oli_unit"]
+        _logger.info(f"Using {conc_unit} for inflows input")
+        for k, v in inflows.items():
+            charge = get_charge(k)
+            input_list.append(
+                {
+                    "group": get_charge_group(charge),
+                    "name": get_oli_name(k),
+                    "unit": self.input_unit_set["molecularConcentration"]["oli_unit"],
+                    "value": v,
+                    "charge": charge,
+                }
+            )
 
-        :param state_vars: dictionary containing solutes, temperatures, pressure, and units
+        json_input = self._add_to_json(
+            "wateranalysis",
+            input_list,
+            included_solids,
+            excluded_solids,
+            use_scaling_rigorous,
+            file_name,
+        )
+        return json_input
+
+    def configure_flash_analysis(
+        self,
+        inflows,
+        flash_method,
+        temperature=None,
+        pressure=None,
+        calculated_variable=None,
+        enthalpy=None,
+        vapor_amount=None,
+        vapor_fraction=None,
+        volume=None,
+        ph=None,
+        acid_titrant=None,
+        base_titrant=None,
+        formed_solid=None,
+        precipitant_inflow=None,
+        included_solids=None,
+        excluded_solids=None,
+        contact_surface=None,
+        flow_type=None,
+        diameter=None,
+        liq_velocity=None,
+        gas_velocity=None,
+        rot_velocity=None,
+        shear_stress=None,
+        roughness=None,
+        nonaqueous_visc=None,
+        water_cut_inversion=None,
+        relative_visc_inversion=None,
+        use_scaling_rigorous=True,
+        file_name=None,
+    ):
 
-        :return input_list: deepcopy of self.water_analysis_input_list
+        """
+        Configure Flash Analysis JSON input.
+
+        :param inflows: dictionary of solutes, of the form {"unit": unit, "values": {solute: concentration}}
+        :param flash_method: string for flash calculation name
+        :param temperature: float for temperature in Kelvins
+        :param pressure: float for pressure in Pascals
+        :param calculated_variable: string for variable to calculate, such as temperature or pressure, used in 'bubblepoint', 'dewpoint', 'vapor-amount', 'vapor-fraction', and 'isochoric' flashes
```

Comment:
Haven't looked that far ahead yet in the PR, but we should make sure to include logic to check this arg if it is not None and ensure that only valid options can be specified. Otherwise, raise an exception stating the erroneous specification by the user and then display the valid options.